src/pages/public_registration/log/LogMessageAccordion.tsx

```diff
@@ -37,17 +50,37 @@ export const LogMessageAccordion = ({ messages, messageBackgroundColor }: LogMes
             p: 0,
             gap: '0.25rem',
           }}>
-          {messages.map((message) => (
-            <MessageItem
-              key={message.identifier}
-              text={message.text}
-              date={message.createdDate}
-              username={message.sender}
-              backgroundColor={messageBackgroundColor}
-            />
-          ))}
+          {messages.map((message) => {
+            const messageType = getTicketTypeFromLogEntryTopic(topic);
+            const canDeleteMessage = !!user && userCanDeleteMessage(user, message, messageType);
+            return (
+              <MessageItem
+                key={message.identifier}
+                text={message.text}
+                date={message.createdDate}
+                username={message.sender}
+                backgroundColor={messageBackgroundColor}
+                menuElement={
+                  canDeleteMessage && (
+                    <MessageMenu messageId={message.id} refetchData={refetchData} disableDelete={!!message.text} />
```

Comment:
Samme som tidligere kommentar

```suggestion
                  canDeleteMessage && (
                    <MessageMenu messageId={message.id} refetchData={refetchData} />
```